### DISCUSSION TITLE
Disable General keymaps

### DISCUSSION THREAD

**Discussion by RiteshChepuri** [LazyVim/LazyVim#2057] [Q&A]
I want to disable the keymaps (General) which are not associated with any plugin. How do I that?

**Comment by abeldekat** [comment]
See the [default settings](https://www.lazyvim.org/configuration#default-settings)

```lua
{
    "LazyVim/LazyVim",
    opts = {
      defaults = {
        keymaps = false,
      },
    },
  }
```

**Reply by RiteshChepuri** [reply]
no its disabling my keymaps. I want to disable the keymaps from lazyvim

**Reply by abeldekat** [reply]
Using stock [LazyVim starter](https://github.com/LazyVim/starter):

`lua/config/keymaps.lua`:

```lua
vim.keymap.set("n", "<leader>l", function()
  vim.print("Personal keymaps are loaded...")
end, { desc = "Personal keymaps are loaded..." })
```

`lua.plugins.init.lua`:

```
return {
  {
    "LazyVim/LazyVim",
    opts = {
      defaults = {
        keymaps = false,
      },
    },
  },
}
```

The result: None of the *general* mappings defined in [LazyVim](https://github.com/LazyVim/LazyVim/blob/main/lua/lazyvim/config/keymaps.lua) are created. The key I created in my own `keymaps.lua` is defined.

Looking at your [dotfiles](https://github.com/RiteshChepuri/dotfiles/blob/77084791d9be0ef40f433cea6c156ba4d2c0e06c/nvim/lua/config/keymaps.lua#L82), I would advise to only define general keys in `lua.config.keymaps.lua`. The keys for plugins are best created inside the plugin specifications defined in `lua.plugins`.

Examples in my own config:

1. [keymaps.lua](https://github.com/abeldekat/nvim/blob/5c3b911bc6dad7a18c4a8094685ce86e73c6d96a/lua/config/keymaps.lua#L1)
2. [editor.lua](https://github.com/abeldekat/nvim/blob/5c3b911bc6dad7a18c4a8094685ce86e73c6d96a/lua/plugins/editor.lua#L69)
